README.md

```diff
@@ -49,45 +64,102 @@ Environment variables:
+**NOTE:** Before running the example below, make sure that you have successfully pushed to the app and that the `Application commit` is the correct version of the code you intend to preload.
 
-Download a OS image from the Resin dashboard and then run with docker:
+Download an OS image from the Resin dashboard and then run:
 
 1) Using option flags for configuration:
 
 ```bash
-./preload.sh --app 123456 --api-token "XXX..." --img /path/to/resin.img
+$ resin-preload --app 123456 --api-token "XXX..." --img /path/to/resin.img
 ```
 
 2) Using environment variables for configuration:
 
 ```bash
 # Set the environment variables
-export API_TOKEN=... # copy from dashboard preferences
-export APP_ID=... # id of your application (you can see it on dashboard URL when you visit your app page)
-export IMAGE=/path/to/resin.img
+$ export API_TOKEN=... # copy from dashboard preferences
+$ export APP_ID=... # id of your application (you can see it on dashboard URL when you visit your app page)
+$ export IMAGE=/path/to/resin.img
 
 # As all required options have been set via environment variables,
 # there's no need to pass any options, just run
-./preload.sh
+$ resin-preload
 ```
 
 The `/path/to/resin.img` file, will now have the latest version of your application preloaded.
 
-## Additional Options
-* `REGISTRY_HOST`: Docker registry from which to download the image. If unsure, use `registry.resin.io`.
+### Additional Options
+
+* `REGISTRY_HOST`: Docker registry from which to download the image. If unsure, leave empty.
 * Alternatively to `API_TOKEN`, you can use an `API_KEY` variable with an API key from a config.json file or the [SDK](https://github.com/resin-io/resin-sdk/blob/master/DOCUMENTATION.md#resin.models.application.getApiKey).
 * `API_HOST`: Address of the Resin API. If unsure, use `https://api.resin.io`.
+
+## Module usage
+
+```js
+const preload = require('resin-preload')
+```
+
+Building the preloader docker image:
+
+```js
+// preload.build() returns a ChildProcess,
+// `spawnOptions` are passed to `childProcess.spawn()`
+var build = preload.build(spawnOptions)
+
+build.once('exit', (code, signal) => {
+  // ...
+})
+```
+
+Preloading a disk image (this requires to have built the preloader image):
+
+```js
+var options = {
+  appId: '123456', // Application ID of the app to preload
+  image: 'resin-RPi3-2.0.0+rev3-4.1.1-...img', // Image to preload into
+  apiToken: 'pZCI6MTc2OCwidXNlc...', // API credentials
+  apiKey: null,
+  apiHost: 'https://api.resin.io', // Optional
+  registryHost: 'registry.resin.io', // Optional
+  containerName: 'resin-image-preloader' // Optional, Docker container name
+}
+
+// preload.run() also returns a ChildProcess,
+// `spawnOptions` are passed to `childProcess.spawn()`
+var run = preload.run(options, spawnOptions)
+
+run.once('exit', (code, signal) => {
+  // ...
+})
+```
+
+## Known Issues
+
+### Flasher Images Unsupported
+
+Currently flasher-type images are unsupported.
```

Comment:
Might be better to clarify that `flasher types are OS images that write the OS to internal media (like an eMMC)`